docs/data-structures.md

```diff
@@ -0,0 +1,249 @@
+# Data Structures of Cell/Script/Transaction/Block
+
+This documents list all the basic structures one may need to know in order to develop on CKB, or mine CKB coins. 
+
+* [Cell](#Cell)
+* [Script](#Script)
+* [Transaction](#Transaction)
+* [Block](#Block)
+
+
+
+## Cell
+
+### Example
+
+```json
+{
+    "capacity": 5000000,
+    "data": "0x",
+    "lock": "0xa58a960b28d6e283546e38740e80142da94f88e88d5114d8dc91312b8da4765a",
+    "type": null
+}
+```
+
+## Description
+
+| Name       | Type       | Description                                                  |
+| :--------- | :--------- | :----------------------------------------------------------- |
+| `capacity` | uint64     | **The size of the cell.** When a new cell is generated (via transaction), one of the verification rule is `capacity = len(capacity)+len(data)+len(type)+len(lock)`. This value also represents the balance of CKB coin, just like the `balance` field in the Bitcoin's UTXO. (E.g. Alice owns 100 CKB coins means she can unlock a group of cells that has 100 amount of `capacity` in total.) |
+| `data`     | Bytes      | **Arbitrary data.** This part is for storing states or scripts.  In order to make this cell valid on-chain, the data filled in this field should comply with the logics and rules defined by `type` or `lock`. |
+| `type`     | `Script`   | **A Script that defines the type of the cell.** In a transaction, if an input cell and an output cell has the same `type` field, then the `data` part of these two cells is limited by the `type` script upon the transaction verification. (I.e. `type` is a script that limits how the `data` field of the new cells can be changed from the old cells.) `type` is required to has a data structure of `script`. This field can be empty in a Cell. |
```

Comment:
The description is not accurate.

@xxuejie Could you help to give a description of `type`?

Reply:
`type` is an option field, it should be explicit to explain.

Reply:
`A script used to enforce certain cell properties. Typical use cases including user defined token support, cell data format validation, etc.`

Reply:
done